src/util/misc.rs

```diff
@@ -34,31 +80,46 @@ pub fn remove_expired(pastas: &mut Vec<Pasta>, args: &Args) {
             // keep
             true
         } else {
+            log::debug!("Removing expired pasta ID: {} ({})", p.id, p.id_as_animals(&args.hash_ids));
+            if p.expiration > 0 && p.expiration <= timenow {
+                log::debug!("  Reason: Expired (expiration: {}, now: {})", p.expiration, timenow);
+            }
+            if p.burn_after_reads > 0 && p.read_count >= p.burn_after_reads {
+                log::debug!("  Reason: Burn after reads limit reached ({}/{})", p.read_count, p.burn_after_reads);
+            }
+            if args.gc_days > 0 && p.last_read_days_ago() >= args.gc_days {
+                log::debug!("  Reason: GC limit reached (last read {} days ago, limit: {})", p.last_read_days_ago(), args.gc_days);
+            }
+            
             // remove from database
```

Comment:
unneeded comment, though I get why you prefer them. As you work on rust longer it gets quite easy to read and you don't need the comments anymore.